### PR TITLE
Public data exhaust API implementation

### PR DESCRIPTION
**Pull request by sowmya-dixit**

### CODE REVIEW

File: analytics-api-core/src/main/scala/org/ekstep/analytics/api/service/JobAPIService.scala [hunk 1]
```diff
@@ -190,16 +254,30 @@ class JobAPIService @Inject()(postgresDBUtil: PostgresDBUtil) extends Actor  {
...
+
+  def _validateDataset(datasetId: String)(implicit config: Config): Map[String, String] = {
+    val validDatasets = config.getStringList("public.data_exhaust.datasets")
+    if (validDatasets.contains(datasetId)) Map("status" -> "true") else Map("status" -> "false", "message" -> s"Provided dataset is invalid. Please use any one from this list - $validDatasets")
+  }
+
+  def getCDNURL(bucket: String, key: String)(implicit config: Config): String = {
+    val cdnHost = config.getString("cdn.host")
+    cdnHost + bucket + "/" + key
+  }
```
Comment (SanthoshVasabhaktula) [inline analytics-api-core/src/main/scala/org/ekstep/analytics/api/service/JobAPIService.scala:282]: CDN url will replace the bucket id as well. Key should be sufficient

File: analytics-api-core/src/main/scala/org/ekstep/analytics/api/service/JobAPIService.scala [hunk 2]
```diff
@@ -76,42 +79,103 @@ class JobAPIService @Inject()(postgresDBUtil: PostgresDBUtil) extends Actor  {
...
       val datasetConfig = if (null != loadConfig.get(datasetId)) loadConfig.get(datasetId).asInstanceOf[java.util.Map[String, AnyRef]] else loadConfig.get("default").asInstanceOf[java.util.Map[String, AnyRef]]
       val bucket = datasetConfig.get("bucket").toString
       val basePrefix = datasetConfig.get("basePrefix").toString
-      val prefix = basePrefix + datasetId + "/" + channel + "/"
+      val prefix = if(channel.isDefined) basePrefix + datasetId + "/" + channel.get + "/" else basePrefix + datasetId + "/"
       APILogger.log("prefix: " + prefix)
 
       val storageKey = config.getString("storage.key.config")
       val storageSecret = config.getString("storage.secret.config")
       val storageService = fc.getStorageService(storageType, storageKey, storageSecret)
```
Comment (SanthoshVasabhaktula) [inline analytics-api-core/src/main/scala/org/ekstep/analytics/api/service/JobAPIService.scala:162]: Storage config for public and private will be different. Also refactor the code a little bit - looks complicated now

**Review by SanthoshVasabhaktula — CHANGES_REQUESTED** [review]

File: analytics-api-core/src/main/scala/org/ekstep/analytics/api/service/JobAPIService.scala (hunk 1 above)
Comment (sowmya-dixit) [reply analytics-api-core/src/main/scala/org/ekstep/analytics/api/service/JobAPIService.scala]: Fixed CDN url by removing bucket.

File: analytics-api-core/src/main/scala/org/ekstep/analytics/api/service/JobAPIService.scala (hunk 2 above)
Comment (sowmya-dixit) [reply analytics-api-core/src/main/scala/org/ekstep/analytics/api/service/JobAPIService.scala]: Added different storage config for public & private.

**Review by SanthoshVasabhaktula — APPROVED** [review]

**Pull request merged**